proof.go

```diff
@@ -42,22 +42,22 @@ func (s *SRP) M(salt []byte, uname string) ([]byte, error) {
 	h := sha256.New()
 
 	if _, err := h.Write(groupHash[:]); err != nil {
-		return nil, fmt.Errorf("failsed to write group hash to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write group hash to hasher: %v", err)
 	}
 	if _, err := h.Write(uHash[:]); err != nil {
-		return nil, fmt.Errorf("failsed to write u hash to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write u hash to hasher: %v", err)
 	}
 	if _, err := h.Write(salt); err != nil {
-		return nil, fmt.Errorf("failsed to write salt to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write salt to hasher: %v", err)
 	}
 	if _, err := h.Write(s.ephemeralPublicA.Bytes()); err != nil {
-		return nil, fmt.Errorf("failsed to write A to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write A to hasher: %v", err)
 	}
 	if _, err := h.Write(s.ephemeralPublicB.Bytes()); err != nil {
-		return nil, fmt.Errorf("failsed to write B to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write B to hasher: %v", err)
 	}
 	if _, err := h.Write(s.key); err != nil {
-		return nil, fmt.Errorf("failsed to write key to hasher: %v", err)
+		return nil, fmt.Errorf("failed to write key to hasher: %v", err)
```

Comment:
Isn't it just lovely when one pastes the same error all over the place? And each with each paste I failsed to notice the typo.